frontend/src/modules/entitydetails/components/Helpers.js

```diff
@@ -55,7 +55,7 @@ export default class Helpers extends React.Component<Props> {
 
         const machineryCount = machinery.translations.length;
         const translations = otherlocales.translations;
-        const otherlocalesCount = translations.other.length || translations.preferred.length;
+        const otherlocalesCount = Object.keys(translations).length === 0;
```

Comment:
`translations` should always have keys, unless we're `fetching`. In practice however, since we abort the request
when switching strings quickly, `translations` might indeed become [`{}`](https://github.com/mozilla/pontoon/blob/1174c7027c91bc85f601d90f2cbed07bfc50a5fb/frontend/src/core/api/base.js#L71).

I've created a patch to store `null` instead of `{}` when request gets aborted. Please download and `git apply` it:
https://gist.github.com/mathjazz/3486ce4c41ffe96e330db046ea4b3de4

The patch also moves the checks from this file to corresponding `Count` modules, so that we don't for example block presenting Machinery count while waiting for Locales.